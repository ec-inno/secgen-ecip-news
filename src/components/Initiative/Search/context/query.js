import React from 'react';

import { defaultLangKey } from '../../../../../languages';

const queryInit = {
  filters: {},
  status: 'ALL',
  language: defaultLangKey,
  pagination: '0/10',
};

const queryReducer = (state, action) => {
  switch (action.type) {
    case 'paginate': {
      return {
        ...state,
        pagination: action.pagination,
      };
    }

    case 'changeStatus': {
      return {
        ...state,
        pagination: queryInit.pagination,
        status: action.status,
      };
    }

    case 'changeLanguage': {
      return {
        ...state,
        pagination: queryInit.pagination,
        language: action.language,
      };
    }

    case 'changeFilter': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          [action.filter]: action.filterValue,
        },
      };
    }

    case 'setFilters': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          ...action.filters,
        },
      };
    }

    case 'removeFilter': {
      const { filters } = state;
      delete filters[action.filter];

      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...filters,
        },
      };
    }

    case 'reset':
    default: {
      return queryInit;
    }
  }
};

const Context = React.createContext();

export { queryInit, queryReducer, Context as default };
