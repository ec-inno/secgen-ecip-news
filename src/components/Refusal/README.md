# Refusal

The component provides feedback on why a given initiative is refused/rejected. Status message translations are based on a CSV file. Follow the guidelines below to update `messages.json` when necessary.

## Using ECI CLI

```sh
yarn eci-cli
```

The command used to convert the input CSV file to JSON is:

```sh
yarn eci-cli convert csvToJson
```

## Parameters and options

- parameters: it's only 1 - the path to the CSV file to convert.
- options: it's also only 1 - the output path

Examples:

```sh
yarn eci-cli convert csvToJson ~/Downloads/rejections.csv --output src/components/Refusal
```

Or specify name:

```sh
yarn eci-cli convert csvToJson ~/Downloads/rejections.csv --output src/components/Refusal/messages.json
```

Or if you want to simply get `output.json` in the root of the project:

```sh
yarn eci-cli convert csvToJson ~/Downloads/rejections.csv
```
