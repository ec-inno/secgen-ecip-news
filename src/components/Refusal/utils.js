import messages from './messages.json';

export const reasonsSupported = [
  'reason.action.registration.reject.abuse',
  'reason.action.registration.reject.competences',
  'reason.action.registration.reject.requirements',
  'reason.action.registration.reject.values',
];

export const getTranslation = ({ code, language }) =>
  messages.find(
    message => message.LNG_CODE === language && message.RTR_KEY === code
  );

export const getReasons = ({ input, language }) => {
  const output = [];

  input.forEach(reason => {
    if (reasonsSupported.includes(reason)) {
      const translation = getTranslation({ code: reason, language });
      if (translation) {
        output.push(translation);
      }
    }
  });

  return output;
};
