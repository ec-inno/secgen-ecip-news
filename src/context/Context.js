import React from 'react';

const initialState = {
  hideOverlay: true,
};

export const SET_LANGUAGE_OVERLAY_VISIBILITY =
  'SET_LANGUAGE_OVERLAY_VISIBILITY';

const reducer = (state, action) => {
  switch (action.type) {
    case SET_LANGUAGE_OVERLAY_VISIBILITY: {
      return Object.assign({}, state, {
        hideOverlay: action.hideOverlay,
      });
    }

    default: {
      return initialState;
    }
  }
};

const Context = React.createContext();

const ContextProviderWithReducer = ({ children }) => {
  const [store, dispatch] = React.useReducer(reducer, initialState);

  return (
    <Context.Provider value={{ store, dispatch }}>{children}</Context.Provider>
  );
};

export { ContextProviderWithReducer, Context as default };
