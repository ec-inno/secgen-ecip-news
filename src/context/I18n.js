import React from 'react';

const I18n = React.createContext();

export default I18n;
