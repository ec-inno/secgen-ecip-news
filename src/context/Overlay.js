import React, { useContext } from 'react';

const Context = React.createContext();

const useOverlayContext = () => useContext(Context);

export { useOverlayContext, Context as default };
