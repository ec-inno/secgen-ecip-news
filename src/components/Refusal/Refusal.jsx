import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

import { useI18nContext } from '@eci/context/I18n';
import { getReasons } from './utils';

const Refusal = ({ reasons: input }) => {
  if (!input || input.length === 0 || !Array.isArray(input)) return '';

  const { locale: language } = useI18nContext();
  const reasons = getReasons({ input, language });

  if (reasons.length === 0) return '';

  const { t } = useTranslation();

  return (
    <div className="eci-answer ecl-u-pa-m ecl-u-mb-l">
      <h2 className="ecl-u-type-heading-2">
        {t(
          "Commission's reply stating the reasons for refusal of registration"
        )}
      </h2>
      <ul className="ecl-u-type-paragraph">
        {reasons.map((reason, key) => (
          <li key={key} className="ecl-u-type-paragraph">
            {reason.RTR_MESSAGE}
          </li>
        ))}
      </ul>
    </div>
  );
};

Refusal.propTypes = {
  reasons: PropTypes.arrayOf(PropTypes.string),
};

export default Refusal;
