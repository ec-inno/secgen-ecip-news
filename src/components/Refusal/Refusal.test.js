import React from 'react';
import renderer from 'react-test-renderer';
import * as I18nContext from '@eci/context/I18n';
import Refusal from './Refusal';

describe('Requirements', () => {
  it('messages.json is present', () => {
    const messages = require('./messages.json');
    expect(messages).not.toBe(undefined);
    expect(Array.isArray(messages)).toBe(true);
    expect(messages.length > 0).toBe(true);
  });

  it('supported reasons are public', () => {
    const { reasonsSupported } = require('./utils');
    expect(reasonsSupported).not.toBe(undefined);
    expect(Array.isArray(reasonsSupported)).toBe(true);
    expect(reasonsSupported.length > 0).toBe(true);
    expect(reasonsSupported).toMatchSnapshot();
  });
});

describe('Refusal', () => {
  it('returns an emptry string on empty input', () => {
    const tree = renderer.create(<Refusal />).toJSON();
    expect(tree).toBe('');
  });

  it('returns an emptry string on empty list of reasons', () => {
    const tree = renderer.create(<Refusal reasons={[]} />).toJSON();
    expect(tree).toBe('');
  });

  it('shows a list of reasons when they are in the list of supported reasons', () => {
    jest.spyOn(I18nContext, 'useI18nContext').mockImplementation(() => ({
      locale: 'en',
    }));

    const tree = renderer
      .create(<Refusal reasons={['reason.action.registration.reject.abuse']} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('respects current language through the locale', () => {
    jest.spyOn(I18nContext, 'useI18nContext').mockImplementation(() => ({
      locale: 'hu',
    }));

    const tree = renderer
      .create(<Refusal reasons={['reason.action.registration.reject.abuse']} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
