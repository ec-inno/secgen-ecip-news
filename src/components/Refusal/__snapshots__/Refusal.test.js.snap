// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Refusal respects current language through the locale 1`] = `
<div
  className="eci-answer ecl-u-pa-m ecl-u-mb-l"
>
  <h2
    className="ecl-u-type-heading-2"
  >
    Commission's reply stating the reasons for refusal of registration
  </h2>
  <ul
    className="ecl-u-type-paragraph"
  >
    <li
      className="ecl-u-type-paragraph"
    >
      a javasolt polgári kezdeményezés egyértelműen visszaélésszerű, komolytalan vagy zaklató jellegű;
    </li>
  </ul>
</div>
`;

exports[`Refusal shows a list of reasons when they are in the list of supported reasons 1`] = `
<div
  className="eci-answer ecl-u-pa-m ecl-u-mb-l"
>
  <h2
    className="ecl-u-type-heading-2"
  >
    Commission's reply stating the reasons for refusal of registration
  </h2>
  <ul
    className="ecl-u-type-paragraph"
  >
    <li
      className="ecl-u-type-paragraph"
    >
      The proposed citizens' initiative is manifestly abusive, frivolous or vexatious.
    </li>
  </ul>
</div>
`;

exports[`Requirements supported reasons are public 1`] = `
Array [
  "reason.action.registration.reject.abuse",
  "reason.action.registration.reject.competences",
  "reason.action.registration.reject.requirements",
  "reason.action.registration.reject.values",
]
`;
